test/main-spec.js

```diff
@@ -12,185 +12,451 @@
 
```

Comment:
I have thought about this a little now: There are a is a lot of duplicate code now in the tests, which also means, that the diff-view of this file is a little messed up (at least here, on github). Also, I think it would be better to keep the tests with separate promise-libraries in separate processes, so that there are no side-effects by loading a library. I would suggest the following: Let the promise library be configurable by an environment-variable like

```js
var TestedPromise;
switch (process.env['PROMISE_LIB']) {
case 'q': TestedPromise = require('q').Promise; break;
...
default: throw new Error('Invalid parameter: ' + process.env['PROMISE_LIB'])
}
```
The promise constructor can be passed to promisedHandlebars . The `delay`-function and the `exists`-function in the test-setup can be implemented like you did in `test/handlebars-setup-es6.js`, using the `TestedPromise`-constructor no matter which library is used. That way, the diff should be clearer.

Finally, the `test`-script in `package.json` could be modified to run `mocha` multiple times with different  with all libraries can be done in the `package.json` (scripts -> test), which could be implemented like:

```
"scripts": {
    "pretest": "standard --version || npm -g install standard",
    "test": "PROMISE_LIB=q mocha && PROMISE_LIB=bluebird mocha && PROMISE_LIB=global mocha && standard",
   ...
}
```

I don't know if this will work for all OS, though. Maybe a commandline-parameter is the better option (like [here](http://stackoverflow.com/questions/16144455/mocha-tests-with-extra-options-or-parameters)).

Maybe you have other ideas, but I'd like to keep the test code clear and with as little duplication as possible.